### ISSUE TITLE
port datasets from the old to the new API

### ISSUE THREAD

**Issue by pmeier** [pytorch/vision#5336]
The new dataset API is now stable enough to start porting more datasets from the old API. For the `0.13.0` release planned for 2022H2 we want to achieve at least feature parity for the new API. If you want to help out, please comment on the respective issue so we can assign it to you.

The process of adding a dataset to the new API is described [here](https://github.com/pytorch/vision/blob/main/torchvision/prototype/datasets/_builtin/README.md). In addition, we already ported some datasets that you could use as reference. In any case, if you are blocked by something feel free to send a partial PR and ping me there so I can help.

The following datasets need to be ported:

## Image classification

Image classification datasets are good starting point if you are not familiar with the dataset or the new API since they these datsets tend to be the easiest.

- [x] #5337
- [x] #5338
- [x] #5339 [^1]
- [x] #5340 [^1]
- [x] #5341
- [x] #5342
- [x] #5343
- [x] #5344 [^3]
- [x] #5345
- [x] #5346
- [x] #5347
- [x] #5348
- [x] #5349 [^1]
- [x] #5350
- [x] #5351
- [x] #5352

## Image detection or segmentation

Image detection or segmentation datasets tend to be a little harder since one needs to merge more infomation into one sample compared to classification. My suggestion is to only pick one of these if you are either familiar with the dataset or the new API so you don't have two manage two things at once.

- [ ] #5353 (@vfdev-5 )
- [x] #5354 [^2]
- [x] #5355 [^2] [^3]
- [x] #5356 [^2]
- [x] #5367

## Image pairs

We are still designing how exactly image pair datasets should be implemented. I list them here for completeness, but I suggest not picking up any of them until the design is finished.

- [x] #5357
- [x] #5358

## Video classification

We are still designing how exactly video datasets should be implemented. I list them here for completeness, but I suggest not picking up any of them until the design is finished.

- [x] #5359
- [x] #5360
- [x] #5361

## Optical flow

We are still designing how exactly optical flow datasets should be implemented. I list them here for completeness, but I suggest not picking up any of them until the design is finished.

- [x] #5362 [^3]
- [x] #5363
- [x] #5364
- [x] #5365
- [x] #5366

[^1]: These datasets do not provide public download links for the data so they might be harder to work on.
[^2]: These datasets are implemented as classification datasets in the old API, but provide extra annotations for detection or segmentation.
[^3]: Maybe we should have `lfw/people`, `kitti/object`, and `kitti/flow` datasets to cleanly separate the different variants. This also applies to `coco` as discussed in https://github.com/pytorch/vision/pull/5326#discussion_r796813705

cc @pmeier @bjuncek

**Comment by Dbhasin1** [comment]
Hi!
I would like to work on #5337, #5338, #5339 for starting out, thanks!

**Comment by pmeier** [comment]
Hey @Dbhasin1, thanks a lot for the interest. I can only assign you to an issue if you comment on it. Could you please do so on the three you picked?

**Comment by abhi-glitchhg** [comment]
I could try for STL10

**Comment by pmeier** [comment]
Hey @Dbhasin1 @abhi-glitchhg @vballoli @Amapocho @vfdev-5. We recently merged #5407, which included some changes to the prototype datasets. I believe the only touching point for you is the removal of the `decoder`. This means, `_make_datapipe` no longer gets passed a `decoder`. Instead you can `from torchvision.prototype.features import EncodedImage` and just use `image = EncodedImage.from_file(buffer)` rather than the old idiom `image = decoder(buffer) if decoder else buffer`.

**Comment by yassineAlouini** [comment]
@pmeier Since food101 is almost done (waiting for the CLA from my employer), I can work on another dataset. Is Kitti available?

**Comment by pmeier** [comment]
@yassineAlouini Yes, Kitti is still open. Please comment on #5355 so I can assign you. That being said, the CLA is a hard requirement and we can't accept contributions without it. So if this just a matter of time you can go ahead, but if there is change the CLA is not approved by your employerr, you might want to wait until the process is finished. Otherwise you might end up with a lot of wasted work.

**Comment by yassineAlouini** [comment]
@pmeier Should be a formality by now, I will just check the old API while waiting for the document. :ok_hand:

**Comment by pmeier** [comment]
Hey everyone. We decided to remove the help "wanted label" for now. Reviewing the PRs takes more time out of my schedule than I anticipated. This is not comment on the quality of the PRs, but rather a late acknowledgement that due to their diverse nature datasets are hard to review. We very much appreciate ever contribution towards closing this issue.

@yassineAlouini @puhuk @abhi-glitchhg @Dbhasin1 @zhiqwang @Amapocho you all have issues assigned to you for which there is no PR yet. If you haven't started yet, I suggest not starting until give another signal here. If you already have an implementation, you might also send a PR and I will try to review them in a timely manner. 

@lezwon @yassineAlouini @vballoli Our decision has no effect on your already open PRs. I'll review them normally.

**Comment by yassineAlouini** [comment]
@pmeier I have started working on the Kitti one but I can pause the work for now and work on something else in the meantime (any suggestion is welcome). And thanks again for your time and guidance, it takes a lot of effort but is very appreicated. 🥰

**Comment by puhuk** [comment]
@pmeier I'm supposed to send PR within this week and I also can pause the work :)
If you want me to stop sending PR, I can stop, or let me send it by this weekend.

**Comment by pmeier** [comment]
@puhuk if you already have some code, you can go ahead and send a PR to not let your work go to waste. Otherwise, pausing development for some time would also be appreciated.

**Comment by puhuk** [comment]
@pmeier Sorry for late reply. Let me pause the development and stop this issue.
